src/udiskslinuxloop.c

```diff
@@ -247,136 +273,43 @@ handle_delete (UDisksLoop            *loop,
         goto out;
     }
 
-  escaped_device = udisks_daemon_util_escape_and_quote (udisks_block_get_device (block));
-
-  if (!udisks_daemon_launch_spawned_job_sync (daemon,
-                                              NULL, /* UDisksObject */
-                                              "loop-setup", caller_uid,
-                                              NULL, /* GCancellable */
-                                              0,    /* uid_t run_as_uid */
-                                              0,    /* uid_t run_as_euid */
-                                              NULL, /* gint *out_status */
-                                              &error_message,
-                                              NULL,  /* input_string */
-                                              "losetup -d %s",
-                                              escaped_device))
+  job = udisks_daemon_launch_simple_job (daemon,
+                                         UDISKS_OBJECT(object),
+                                         "loop-setup",
+                                         caller_uid,
+                                         NULL);
+
+  if (job == NULL)
     {
-      g_dbus_method_invocation_return_error (invocation,
-                                             UDISKS_ERROR,
-                                             UDISKS_ERROR_FAILED,
-                                             "Error deleting %s: %s",
-                                             udisks_block_get_device (block),
-                                             error_message);
+      g_dbus_method_invocation_return_error (invocation, UDISKS_ERROR, UDISKS_ERROR_FAILED,
+                                             "Failed to create a job object");
       goto out;
     }
 
+  if (!bd_loop_teardown (udisks_block_get_device (block), &error))
+    {
+      g_prefix_error (&error, "Error deleting %s: ", udisks_block_get_device (block));
```

Comment:
This and the above should actually use ``udisks_block_dup_device()`` (and then ``g_free()`` the value when not needed) because this is code runs in a non-main thread and the value could be changed by a different thread in the meantime.